.github/workflows/publish_pypi.yml

```diff
@@ -212,6 +213,34 @@ jobs:
           path: ./dist/*.tar.gz
           if-no-files-found: error
 
+  publish_anaconda:
+    # This job is only of value to PyBaMM and would always be skipped in forks
+    # It is meant to run on a schedule, when manually triggered, and does not
+    # run on releases.
+    if: >-
+      (github.repository == 'pybamm-team/PyBaMM') &&
+      (github.event_name == 'workflow_dispatch' && github.event.inputs.target == 'anaconda') ||
+      (github.event_name == 'schedule')
+    name: Upload nightly wheels to Anaconda.org PyPI index
+    needs: [build_macos_and_linux_wheels, build_macos_arm64_wheels, build_windows_wheels]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download all artifacts
+        uses: actions/download-artifact@v4
+
+      - name: Move all package files to files/
+        run: |
+          mkdir files
+          mv windows_wheels/* linux_wheels/* macos_amd64_wheels/* macos_arm64_wheels/* files/
+
+      - name: Push to Anaconda PyPI index
+        uses: scientific-python/upload-nightly-action@b67d7fcc0396e1128a474d1ab2b48aa94680f9fc  # v0.5.0
```

Comment:
It's recommended to pin this action to a commit hash. I think we should do this for all of our other actions in other jobs too.